apps/labs/posts/pep-517-build-system-popularity.md

```diff
@@ -0,0 +1,615 @@
+
+The three next top ranking build system families are Poetry (8.4% packages),
+Hatchling (6.5% packages) and Flit (3.8%). Build systems other than the four
+already mentioned amount for 2.6% of packages. They include both generic Python build
+systems (such as `pdm-backend`), as well as more specialized tools such as:
+
+- Maturin — used to build Rust packages (1.1%)
+
+- scikit-build-core — used to integrate with CMake build system (0.40%)
+
+- meson-python — used to integrate with Meson build system (0.22%)
+
+All of setuptools, Poetry and Hatchling support plugins. Packages that
+need to extend their behavior usually use the same standard build backend
+and enable plugins in their configuration. The exception to that
+is the `poetry_dynamic_versioning` plugin that uses a separate backend.
+Conversely, `flit_core` does not support plugins and is extended
+by creating new backends based on it. Two examples of such backends
+are `flit_scm` and `flit_gettext`.
+
+Some of the listed PEP 517 backends are derived
+from earlier setuptools plugins. For example, `scikit-build-core` replaces
+the earlier `scikit-build` plugin, and `pbr` provides a setuptools
+plugin along with the newer PEP 517 backend. We can expect the numbers
+corresponding to these backends to grow once packages are updated
+from the older approach of using plugins to the newer approach of using
+the respective backend.
+
+23 source distributions provided a custom backend as a Python module
+inside the distribution. However, only three were truly custom
+and the remainder simply extended another build system. In three out of
+four cases, that build system were setuptools.
+
+19 packages declared the use of `setuptools.build_meta:__legacy__`
+backend. Arguably, [using the legacy backend explicitly is incorrect](https://github.com/pypa/setuptools/issues/1689). 51 packages still used
+the deprecated `poetry.masonry.api` backend, and the deprecated
+`flit.buildapi` and `pdm.pep517.api` backends were used by four packages
+each.
+
+## Different setuptools configuration formats
+
+Setuptools support three different configuration formats right now:
+`setup.py`, `setup.cfg` and `pyproject.toml`.
+
+Configuring via `setup.py` is the oldest approach. It follows the format
+originally used by the earlier `distutils` build system. When the package
+is built, the `setup.py` script is executed with specific commands to
+build a wheel. The script eventually calls the `setup()` function
+provided by setuptools, passing the package metadata and build
+configuration as arguments.
+
+This configuration method is the most flexible, as it permits executing
+any Python code during the build. However, this comes at a price — it is
+easy to make mistakes such as:
+
+```python
+install_requires = []
+# WRONG!
+if sys.version_info < (3, 11):
+    install_requires.append("exceptiongroup")
+```
+
+With the above snippet, if a wheel is built with Python 3.10, it will
```

Comment:
At first I thought you meant sdist, not wheel, here. However, now I think it is wheel but `py3-none-any` wheels specifically. Correct? It reads a bit confusing to me, because I was thinking about `cp31x` wheels.

Reply:
Yes, I meant pure Python wheels. I wish the word "universal" wouldn't be used already for py2/py3 wheel, but perhaps it'd not be too technical to say `py3-*` wheel here.

Reply:
Maybe `universal (py3-*)`?

Or otherwise "a wheel from pure Python code"?